### ISSUE TITLE
about update parameter ?

### ISSUE THREAD

**Issue by ranjiewwen** [kevinlu1211/deep-photo-aesthetics#3]
-  first thans for sharing you code !
- there is one question , in your train code : 

`ignored_params = list(map(id, model.attribute_weights.parameters()))
    base_params = filter(lambda p: id(p) not in ignored_params,
                         model.parameters())
`
`
    optimizer = torch.optim.Adam([
        {'params': base_params},
        {'params': model.attribute_weights.parameters(), 'lr': fc_lr, 'weight_decay': 1e-5}
    ], lr=fine_tune_lr)
`
 -  why you write this way?  there is difference like follow : 
`    optimizer = torch.optim.Adam([
        {'params': model.parameters(), 'lr': fc_lr, 'weight_decay': 1e-5}
    ], lr=fine_tune_lr)`

- can you help me ?thanks again !

**Comment by ranjiewwen** [comment]
i have understood this way, two step use different learning rate . 

another question is : 
` # mashed_loss = loss_by_attribute * weights
            # masked_loss = torch.sum(masked_loss)
            # masked_loss.backward()`
the type different match when use "*" .

**Comment by kevinlu1211** [comment]
Yes, you have understood it correctly, the reason why there are 2 learning rates is because we want to fine-tune the parameters in the CNN layers (which are already tuned to extract features from the image), but for the FC prediction layers they are randomly initialized, therefore they need a larger learning rate.

I don't understand your second question, but the reason we multiple the losses by a weight vector is to give some attributes more weight. For example, maybe we want to weight vivid color more heavily, then we would give the weight corresponding to vivid color a higher weight than the others.
